File: Codeforces/Contests/Educational Codeforces Round 129 (Rated for Div. 2)/Double_Sort.cpp
#include <bits/stdc++.h>


using namespace std;

int main() {

    int t;
    cin>>t;
    while(t--) {
        int n;
        cin>>n;
        vector<int> arr1(n);
        vector<int> arr2(n);
        vector<vector<int>> arr(n, vector<int>(2));
        vector<vector<int>> sortedArr(n, vector<int>(2));
        for (int i = 0; i < n; i++) {
            cin>>arr1[i];
        }
        for (int i = 0; i < n; i++) {
            cin>>arr2[i];
        }
        for (int i = 0; i < n; i++) {
            arr[i][0] = arr1[i];
            arr[i][1] = arr2[i];
            sortedArr[i][0] = arr1[i];
            sortedArr[i][1] = arr2[i];
        }

        sort(sortedArr.begin(), sortedArr.end());

        vector<int> postSortArr1(n);
        vector<int> sortedPostSortArr1(n);
        vector<int> postSortArr2(n);
        vector<int> sortedPostSortArr2(n);

        for (int i = 0; i < n; i++) {
            postSortArr1[i] = sortedArr[i][0];
            sortedPostSortArr1[i] = sortedArr[i][0];
            postSortArr2[i] = sortedArr[i][1];
            sortedPostSortArr2[i] = sortedArr[i][1];
        }

        sort(sortedPostSortArr1.begin(), sortedPostSortArr1.end());
        sort(sortedPostSortArr2.begin(), sortedPostSortArr2.end());

        bool impossible = true;

        for (int i = 0; i < n; i++) {
            if ((sortedPostSortArr1[i] != postSortArr1[i])||(sortedPostSortArr2[i] != postSortArr2[i])) {
                
            }
        }
        
        

        
    }


}

File: Codeforces/Contests/Educational Codeforces Round 129 (Rated for Div. 2)/Double_Sort.py
t = int(input(''))

problems = []

for i in range(t):
    n = int(input(''))
    arr1 = [int(i) for i in input('').split(' ')]
    arr2 = [int(i) for i in input('').split(' ')]
    problems.append((n, arr1, arr2))

for n, arr1, arr2 in problems:
    zip_arr = []
    for i in range(n):
        zip_arr.append([arr1[i], arr2[i], i])
    zip_sorted_arr = zip_arr.sort()

    impossible = False

    for i in range(n):
        if i == 0:
            continue
        if zip_sorted_arr[i][0] >= zip_sorted_arr[i-1][0] and zip_sorted_arr[i][1] >= zip_sorted_arr[i-1][1]:
            continue
        else:
            print(-1)
            impossible = True
            break
    
    if impossible == True:
        continue

    swap_arr = []
    zip_swap_arr = []
    for i in zip_arr:
        swap_arr.append(i[2])
    for i in zip_sorted_arr:
        zip_swap_arr.append(i[2])
    
    swaps = 0
    swap_steps = []

    for i in range(0, n):
        if swap_arr[i] == zip_swap_arr[i]:
            continue

        swap_arr[i], swap_arr[zip_swap_arr[i]] = swap_arr[zip_swap_arr[i]], swap_arr[i]
        swaps += 1
        swap_steps.append((i, zip_swap_arr[i]))

    print(swaps)
    for i, j in swap_steps:
        print(i+1, j+1)
